account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt

```diff
@@ -234,22 +275,56 @@ class NativeTEKStamperModule(reactContext: ReactApplicationContext) :
         val pubSpec = ECPublicKeySpec(bcSpec.g.multiply(s).normalize(), bcSpec)
         val keyFactory =
             KeyFactory.getInstance("EC", BouncyCastleProvider.PROVIDER_NAME)
+
         val ecPublicKey = EllipticCurves.getEcPublicKey(keyFactory.generatePublic(pubSpec).encoded)
 
         // verify the key pair
         EllipticCurves.validatePublicKey(ecPublicKey, ecPrivateKey)
 
-        return Pair(ecPublicKey.encoded, ecPrivateKey.encoded)
+        // compress it to match turnkey expectations
+        val compressedPublicKey = EllipticCurves.pointEncode(
+            EllipticCurves.CurveType.NIST_P256,
+            EllipticCurves.PointFormatType.COMPRESSED,
+            ecPublicKey.w
+        )
+        return Pair(compressedPublicKey, privateKey)
     }
 
-    private fun publicKeyToHex(keyHandle: KeysetHandle): String {
-        val outputStream = ByteArrayOutputStream()
-        keyHandle.publicKeysetHandle.writeNoSecret(
-            BinaryKeysetWriter.withOutputStream(
-                outputStream
+    private fun tekPublicKeyHex(keyHandle: KeysetHandle): String {
+        val keySet = CleartextKeysetHandle.getKeyset(keyHandle.publicKeysetHandle)
+        val hpkePublicKey = HpkePublicKey.parseFrom(keySet.keyList[0].keyData.value)
+
+        val publicKeyBytes = hpkePublicKey.publicKey.toByteArray()
+        return DatatypeConverter.printHexBinary(publicKeyBytes)
+    }
+
+    private fun getHpkePrivateKeyFromKeysetHandle(keysetHandle: KeysetHandle): HpkePrivateKey {
+        val pkKs = CleartextKeysetHandle.getKeyset(keysetHandle)
+        val pkKeyData = pkKs.keyList[0].keyData
+        if (pkKeyData.typeUrl != "type.googleapis.com/google.crypto.tink.HpkePrivateKey") {
+            throw Error("invalid key type")
+        }
+
+        return HpkePrivateKey.create(
+            com.google.crypto.tink.hybrid.HpkePublicKey.create(
```

Comment:
we use the proto version of the public key elsewhere hence the need for this specific one. I'm going to look into refactoring things so we can use the same interfaces throughout